sriovnet_switchdev_test.go

```diff
@@ -11,49 +11,114 @@ import (
 	utilfs "github.com/Mellanox/sriovnet/pkg/utils/filesystem"
 )
 
+type repContext struct {
+	Name         string // create files /sys/bus/pci/devices/<vf addr>/physfn/net/<Name> , /sys/class/net/<Name>
+	PhysPortName string // conditionally create if string is empty under /sys/class/net/<Name>/phys_port_name
+	PhysSwitchID string // conditionally create if string is empty under /sys/class/net/<Name>/phys_switch_id
+}
+
+func setUpRepresentorLayout(vfPciAddress string, rep *repContext) error {
+	path := filepath.Join(PciSysDir, vfPciAddress, "physfn/net", rep.Name)
+	err := utilfs.Fs.MkdirAll(path, os.FileMode(0755))
+	if err != nil {
+		return err
+	}
+
+	path = filepath.Join(NetSysDir, rep.Name)
+	err = utilfs.Fs.MkdirAll(path, os.FileMode(0755))
+	if err != nil {
+		return err
+	}
+
+	if rep.PhysPortName != "" {
+		physPortNamePath := filepath.Join(NetSysDir, rep.Name, netdevPhysPortName)
+		physPortNameFile, _ := utilfs.Fs.Create(physPortNamePath)
+		_, err = physPortNameFile.Write([]byte(rep.PhysPortName))
+		if err != nil {
+			return err
+		}
+	}
+
+	if rep.PhysSwitchID != "" {
+		physSwitchIDPath := filepath.Join(NetSysDir, rep.Name, netdevPhysSwitchID)
+		physSwitchIDFile, _ := utilfs.Fs.Create(physSwitchIDPath)
+		_, err = physSwitchIDFile.Write([]byte(rep.PhysSwitchID))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 //nolint:unparam
-func setupUplinkRepresentorEnv(t *testing.T, vfPciAddress string) func() {
+func setupUplinkRepresentorEnv(t *testing.T, uplink *repContext, vfPciAddress string, vfReps []*repContext) func() {
 	var err error
 	utilfs.Fs = utilfs.NewFakeFs()
-	path := filepath.Join(PciSysDir, vfPciAddress, "physfn/net", "eth0")
+	path := filepath.Join(PciSysDir, vfPciAddress, "physfn/net")
```

Comment:
i see you call this in L#21, do you need it here as well ?

Reply:
It is not needed, removed